### ISSUE TITLE
Integrate `estimate_ctf`, `CTFFilter`, and `CoordinateSource`s

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#622]
#### CTF Estimator

The `estimate_ctf` method (which creates a `CTFEstimator` object internally) takes in a directory containing micrographs and outputs `results`, a list of tuples containing CTF parameters for each micrograph:

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/c2cd98fa03b6bd232472a29c7aa0876d041772b2/src/aspire/ctf/ctf_estimator.py#L697-L713

The first 3 items here correspond to `defocus_u`, `defocus_v`, and `defocus_angle`. 

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/c2cd98fa03b6bd232472a29c7aa0876d041772b2/src/aspire/ctf/ctf_estimator.py#L819-L828

#### CTF Filter

CTFFilter initialization has a different order, but this should be easily fixed. 

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/c2cd98fa03b6bd232472a29c7aa0876d041772b2/src/aspire/operators/filters.py#L374-L387

#### Coordinate Sources

`BoxesCoordinateSource` and `CentersCoordinateSource` objects are initialized with empty `unique_filters` (this variable is designed to store CTF filters)

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/c2cd98fa03b6bd232472a29c7aa0876d041772b2/src/aspire/source/coordinates.py#L126-L133

**Comment by chris-langfield** [comment]
The following code in `RelionSource` extracts CTF parameters from the STAR file and adds them to `unique_filters`. We could probably do something similar for coordinate sources, except that we also need to add the metadata manually using `ImageSource.set_metadata()`. (`RelionSource` gets it directly from the STAR file). 

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/c2cd98fa03b6bd232472a29c7aa0876d041772b2/src/aspire/source/relion.py#L143-L166

**Comment by chris-langfield** [comment]
The question would be how to design the API. 

For instance, should there be a `CoordinateSource.estimate_ctf(pixel_size, voltage ...)`, which then internally calls `aspire.ctf.estimate_ctf()`, and creates the filters and updates the metadata internally? The advantage of this is that the `CoordinateSource` knows the names and locations of all of its micrographs, so would be able to call `aspire.ctf.estimate_ctf()` in a controlled manner. 

Alternatively, there could be a `CoordinateSource` function that accepts the output of `estimate_ctf`, i.e. `results`, and then goes on to create the filters and update the metadata. The disadvantage here is that we would have to do name matching between the micrographs given to `estimate_ctf` and the micrographs stored in the `CoordinateSource`. 

A difficulty that comes up regardless of the API is just the number of parameters that go into `estimate_ctf`. I'm not sure how many of those could be given reasonable defaults. This would be particularly challenging for the `extract-particles` commandline interface to `CoordinateSource`. It would be good to automatically estimate the CTF as this information is needed for many stages of the pipeline. However, would we need to add a command line option for each CTF estimation parameter?

**Comment by chris-langfield** [comment]
@janden and I also discussed that `result`, the output of `estimate_ctf`, should be converted into a dict of dicts. That is:

```
}
      "micrograph_name1" : {"pixel_size": 1, "voltage": 300 ...},
      "micrograph_name2" : {"pixel_size": 1, "voltage": 300 ...},
       ....
}
```

**Comment by chris-langfield** [comment]
Closed by #623
